[docs] Add Grid version 2 docs (#33554)
mui/material-ui: docs/data/material/migration/migration-v4/v5-component-changes.md

});
```

> ### Change negative margin

Reviewer: ```suggestion
### Change negative margins
```

> The negative margin applies only to the top and left sides of the grid container. If you want to preserve the same behavior (negative margin on all sides), we recommend to use the new Grid v2 instead.

Reviewer: ```suggestion
The negative margins apply only to the top and left sides of the grid container. 
If you need negative margins on all sides, we recommend using the new Grid v2 instead:
```

```diff
- import Grid from '@mui/material/Grid';
+ import Grid from '@mui/material/Unstable_Grid2';
```

> To learn more about the Grid v2, check out the [demos](/material-ui/react-grid2/#whats-changed) and the [migration guide](/material-ui/migration/migration-grid-v2/).

Reviewer: ```suggestion
To learn more about the Grid v2, check out the [demos](/material-ui/react-grid2/#whats-changed) and the [Grid migration guide](/material-ui/migration/migration-grid-v2/).
```

:::info

> Grid v2 is introduced in Material UI v5.9.1 which comes with negative margin on all sides by default.

Reviewer: ```suggestion
Grid v2 was introduced in Material UI v5.9.1 and features negative margins on all sides by default.
```
